srml/assets/src/lib.rs

```diff
@@ -14,7 +14,135 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
+//! # Assets Module
+//! <!-- Original author of paragraph: @gavofyork -->
+//!
 //! A simple, secure module for dealing with fungible assets.
+//!
+//! ## Overview
+//!
+//! The assets module provides functionality for asset management of fungible asset classes with a fixed supply, including:
+//!
+//! * Asset Issuance
+//! * Asset Transfer
+//! * Asset Destruction
+//!
+//! To use it in your module, you need to implement the assets [`Trait`].<br />
+//! The supported dispatchable functions are documented in the [`Call`] enum.
+//!
+//! ## Terminology
+//!
+//! * **Asset issuance:** The process of an account issuing a total fixed supply of a new asset class.
+//! * **Asset transfer:** The process of an account transfering units of their holding of an asset to a recipient.
+//! * **Asset destruction:** The process of an account destroying their entire holding of an asset.
+//! * **Fungible asset:** An asset that may be interchanged into an identical equivalent.
+//! * **Non-fungible asset:** An asset that is scarce and offers unique characteristics.
+//!
+//! ## Goals
+//! <!-- Original inspiration of paragraph: @gavofyork / staking module documentation -->
+//! <!-- FIXME - assumptions only. require an expert to peer review (or re-write) -->
+//!
+//! The assets system in Substrate is designed to achieve the following goals:
+//! * It should be possible to create a unique fungible asset.
+//! * It should be possible to issue fungible assets that are controlled by a cold wallet.
+//! * It should be possible to transfer fungible assets between cold wallets.
+//! * It should be possible to destroy a proportion of fungible assets that are controlled by a cold wallet.
+//!
+//! ## Interface
+//!
+//! ### Supported Origins
+//!
+//! **signed** - Used to issue, transfer, and destroy an asset holding.
+//!
+//! ### Dispatchable Functions ([`Call`])
+//!
+//! * `issue` - Issues the total supply of a new fungible asset to the account of the caller of the function.
+//! * `transfer` - Transfers an `amount` of units of a fungible asset `id` from the balance of the sender's account (`origin`) that called the function to a `target` account.
+//! * `destroy` - Destroys the entire holding of a fungible asset `id` associated with the account that called the function from its total supply.
```

Comment:
Doesn't make sense to say "destroy entire X from total" - if it's entire then there is nothing left.
```suggestion
//! * `destroy` - Destroys the entire holding of a fungible asset `id` associated with the account that called the function.
```